### PR TITLE
changes for Enabling Kai via workspace creation

### PR DESCRIPTION
**Pull request by atoshniwal5**
This PR contains changes to enable SingleStore Kai using terraform provider while workspace creation

### CODE REVIEW

**Comment by atoshniwal5** [comment]
@noprysk-ua changes are completed. Please review

**Comment by noprysk-ua** [comment]
Make sure to fix the failed CI https://github.com/singlestore-labs/terraform-provider-singlestoredb/actions/runs/10352270575/job/28954845087?pr=29 (it looks like it's merely a formatting issue).

File: internal/provider/workspaces/resource.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ type workspaceResourceModel struct {
 	Suspended        types.Bool   `tfsdk:"suspended"`
 	CreatedAt        types.String `tfsdk:"created_at"`
 	Endpoint         types.String `tfsdk:"endpoint"`
+	EnableKai        types.Bool   `tfsdk:"kai_enabled"`
```
Comment (noprysk-ua) [inline internal/provider/workspaces/resource.go:45]: EnableKai -> KaiEnabled

File: internal/provider/workspaces/resource.go [hunk 2]
```diff
@@ -288,6 +294,13 @@ func (r *workspaceResource) ModifyPlan(ctx context.Context, req resource.ModifyP
 		return
 	}
 
+	if !plan.EnableKai.Equal(state.EnableKai) {
+		resp.Diagnostics.AddError("Cannot enable kai for workspace",
```
Comment (noprysk-ua) [inline internal/provider/workspaces/resource.go:298]: Cannot change the kai_enabled configuration for the workspace

File: internal/provider/workspaces/resource.go [hunk 3]
```diff
@@ -288,6 +294,13 @@ func (r *workspaceResource) ModifyPlan(ctx context.Context, req resource.ModifyP
...
+			"To prevent accidental deletion of the databases that are attached to the workspace, enabling kai is not permitted.")
```
Comment (noprysk-ua) [inline internal/provider/workspaces/resource.go:299]: Changing the kai_enabled configuration is currently not supported.

**Comment by noprysk-ua** [comment]
Is see the previous pipeline failed for you, make sure to fix it: https://github.com/singlestore-labs/terraform-provider-singlestoredb/actions/runs/10491547756/job/29062271734

**Comment by noprysk-ua** [comment]
I see end-to-end tests failed with 

```
        Error: Provider produced inconsistent result after apply
        
        When applying changes to singlestoredb_workspace.this, provider
        "provider[\"registry.terraform.io/hashicorp/singlestoredb\"]" produced an
        unexpected new value: .kai_enabled: was cty.True, but now **null.**
```

after suspending.

Try adding

```
				PlanModifiers: []planmodifier.String{
					stringplanmodifier.UseStateForUnknown(),
				},
				Computed:            true,
```

**Comment by atoshniwal5** [comment]
> I see end-to-end tests failed with
> 
> ```
>         Error: Provider produced inconsistent result after apply
>         
>         When applying changes to singlestoredb_workspace.this, provider
>         "provider[\"registry.terraform.io/hashicorp/singlestoredb\"]" produced an
>         unexpected new value: .kai_enabled: was cty.True, but now **null.**
> ```
> 
> after suspending.
> 
> Try adding
> 
> ```
> 				PlanModifiers: []planmodifier.Bool{
> 					boolplanmodifier.UseStateForUnknown(),
> 				}, // for your new resource attribute
> ```

It still fails after this.

**Comment by noprysk-ua** [comment]
@atoshniwal5 I've ran the pipeline with the change and it worked. Just apply this change
```
diff --git a/internal/provider/workspaces/resource.go b/internal/provider/workspaces/resource.go
index e0f290b..07d5f42 100644
--- a/internal/provider/workspaces/resource.go
+++ b/internal/provider/workspaces/resource.go
@@ -9,6 +9,7 @@ import (
        "github.com/hashicorp/terraform-plugin-framework/resource"
        "github.com/hashicorp/terraform-plugin-framework/resource/schema"
        "github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+       "github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
        "github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
        "github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
        "github.com/hashicorp/terraform-plugin-framework/schema/validator"
@@ -99,6 +100,9 @@ func (r *workspaceResource) Schema(_ context.Context, _ resource.SchemaRequest,
                                MarkdownDescription: "The endpoint used to connect to the workspace.",
                        },
                        "kai_enabled": schema.BoolAttribute{
+                               PlanModifiers: []planmodifier.Bool{
+                                       boolplanmodifier.UseStateForUnknown(),
+                               },
                                Optional:            true,
                                MarkdownDescription: "Whether the Kai API is enabled for the workspace.",
                        },

```

**Comment by noprysk-ua** [comment]
Moving to https://github.com/singlestore-labs/terraform-provider-singlestoredb/pull/30

**Pull request closed without merging**